src/wp-includes/media.php

```diff
@@ -4096,7 +4096,21 @@ function wp_get_image_editor( $path, $args = array() ) {
  * @return bool True if an eligible editor is found; false otherwise.
  */
 function wp_image_editor_supports( $args = array() ) {
-	return (bool) _wp_image_editor_choose( $args );
+	$supports = wp_cache_get( 'wp_image_editor_supports', 'image_editor' );
+
+	if ( false === $supports ) {
+		$supports = array();
+	}
+
+	$cache_key = md5( serialize( $args ) );
```

Comment:
I think the note from my previous comment https://github.com/WordPress/wordpress-develop/pull/6936#pullrequestreview-2221590113 is relevant here: Caching `$args` alone is not enough to ensure this cache doesn't serve stale values. We need to also account for the `wp_image_editors` filter output, otherwise someone that installs a plugin that uses this filter won't see the effects. For example, that filter is used in the "Image Placeholders" plugin we have.

I would also argue that caching within `_wp_image_editor_choose()` makes more sense as that function is called in other places too, and the way that it works doesn't change for these other callers. So by caching the result of `_wp_image_editor_choose()` we expand the reach of this enhancement.

In any case, the main problem here is not accounting for the filter in the cache key.

Reply:
[6056c51](https://github.com/WordPress/wordpress-develop/pull/6936/commits/6056c51f2879bbdd63f0b289de90d1b356d0294e)